# these are only used for ICP and reading and reference DP are not changed

readingDataPointsFilters:
  - RandomSamplingDataPointsFilter:
      prob: 1.0
#  - SamplingSurfaceNormalDataPointsFilter:
#      knn: 10

referenceDataPointsFilters:
#  - MaxDistDataPointsFilter:
#      maxDist: 20.0
  - RandomSamplingDataPointsFilter:
      prob: 0.8
#  - SamplingSurfaceNormalDataPointsFilter:
#      knn: 10

matcher:
  KDTreeMatcher:
    knn: 3
    maxDist: 2.0
    epsilon: 1.0

outlierFilters:
  - TrimmedDistOutlierFilter:
     ratio: 0.7
  - SurfaceNormalOutlierFilter:
     maxAngle: 0.5

errorMinimizer:
  PointToPlaneErrorMinimizer

transformationCheckers:
  - DifferentialTransformationChecker:
      minDiffRotErr: 0.001
      minDiffTransErr: 0.01
      smoothLength: 2
  - CounterTransformationChecker:
      maxIterationCount: 40
  - BoundTransformationChecker:
      maxRotationNorm: 0.8
      maxTranslationNorm: 1.2

inspector:
#  VTKFileInspector
  NullInspector

logger:
#  FileLogger
  NullLogger
